### PR TITLE
[DVT-830] RPC Conformance and Fuzz Summary Table and Exporters

### PR DESCRIPTION
**Pull request by IdrisHanafi**
# Description

Added tabular summary for the RPC conformance and fuzz tests. Also included different types of test result exporters such as CSV, JSON, Markdown, and HTML.

## Jira / Linear Tickets

- [DVT-830](https://polygon.atlassian.net/browse/DVT-830?atlOrigin=eyJpIjoiZWM1N2JhOTNiZWMwNGIyOWFlZDk0OGFlNjZiZTExNWYiLCJwIjoiaiJ9)

# Testing

Sample run against geth client:
```
$ time go run main.go rpcfuzz http://localhost:8545 --fuzz --fuzzn 10 --namespaces net,web3 --json-export-path testout/test.json --csv-export-path testout/test.csv --markdown-export-path testout/test.md --html-export-path testout/test.html

{"level":"info","ethAddress":"0x85dA99c8a7C2C95964c8EfD687E95E632Fc533D6","time":"2023-07-12T00:49:06-04:00","message":"Loaded private key"}
{"level":"info","namespaces":["net_","web3_"],"time":"2023-07-12T00:49:06-04:00","message":"enabling namespaces"}
12:49AM INF Running with fuzzed args method=net_version
12:49AM INF Running with fuzzed args method=web3_clientVersion
12:49AM INF Running with fuzzed args method=web3_sha3
12:49AM INF Running with fuzzed args method=web3_sha3
12:49AM INF Running with fuzzed args method=net_listening
12:49AM INF Running with fuzzed args method=net_peerCount
 NAME                             METHOD              TEST(S) PASSED  TEST(S) RAN
 RPCTestNetListening              net_listening                    1            1
 RPCTestNetListening-FUZZED       net_listening                   10           10
 RPCTestNetPeerCount              net_peerCount                    1            1
 RPCTestNetPeerCount-FUZZED       net_peerCount                   10           10
 RPCTestNetVersion                net_version                      1            1
 RPCTestNetVersion-FUZZED         net_version                     10           10
 RPCTestWeb3ClientVersion         web3_clientVersion               1            1
 RPCTestWeb3ClientVersion-FUZZED  web3_clientVersion              10           10
 RPCTestWeb3SHA3                  web3_sha3                        1            1
 RPCTestWeb3SHA3-FUZZED           web3_sha3                        4           10
 RPCTestWeb3SHA3Error             web3_sha3                        1            1
go run main.go rpcfuzz http://localhost:8545 --fuzz --fuzzn 10 --namespaces    2.55s user 1.06s system 122% cpu 2.947 total
```

With some colors too:
Screenshot 2023-07-12 at 12 55 38 AM

- [x] Exports to JSON, CSV, Markdown, and HTML
- [x] Table prints out properly

[DVT-830]: https://polygon.atlassian.net/browse/DVT-830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by leovct — CHANGES_REQUESTED** [review]
lgtm, left a note regarding export path flags :)

File: cmd/rpcfuzz/rpcfuzz.go [hunk 1]
```diff
@@ -139,19 +139,23 @@ const (
...
+	usage                  string
+	testPrivateHexKey      *string
+	testContractAddress    *string
+	testPrivateKey         *ecdsa.PrivateKey
+	testEthAddress         ethcommon.Address
+	testNamespaces         *string
+	testFuzz               *bool
+	testFuzzNum            *int
+	seed                   *int64
+	testJSONExportPath     *string
```
[the remark this answers is not in the document]
Comment (IdrisHanafi) [reply cmd/rpcfuzz/rpcfuzz.go]: updated!

File: cmd/rpcfuzz/rpcfuzz.go [hunk 2]
```diff
@@ -1879,9 +1898,14 @@ func init() {
 	testFuzz = flagSet.Bool("fuzz", false, "Flag to indicate whether to fuzz input or not.")
 	testFuzzNum = flagSet.Int("fuzzn", 100, "Number of times to run the fuzzer per test.")
 	seed = flagSet.Int64("seed", 123456, "A seed for generating random values within the fuzzer")
+	testOutputExportPath = flagSet.String("export-path", "", "The directory export path of the output of the tests. Must pair this with either --json, --csv, --md, or --html")
```
Comment (leovct) [inline cmd/rpcfuzz/rpcfuzz.go:1901]: Just a question, is there a default output mode or the command will fail if the user forgets to specify one of the format flags (csv, html, etc.)?

**Review by leovct — APPROVED** [review]
🥳

File: cmd/rpcfuzz/rpcfuzz.go (hunk 2 above)
Comment (IdrisHanafi) [reply cmd/rpcfuzz/rpcfuzz.go:1906]: good question, for now I've just 1) inform the user on the flag usage level and 2) added a warn message if the export path is present without an export type

**Pull request merged**